### ISSUE TITLE
OpenAPI security

### ISSUE THREAD

**Issue by Azzerty23** [zenstackhq/zenstack#321]
**Is your feature request related to a problem? Please describe.**
As I'm transitioning from FastAPI to Zenstack, I'm used to have the generated Swagger UI with authentication built-in. For endpoints requiring auth, there is a padlock to identify them. I would love to have the same within Zenstack.

**Describe the solution you'd like**
Ideally, I would like an automatic addition of an empty security field in the generated schema when the model/operation (endpoint) doesn't require any authentication (e.g. for a model with the `@@allow('create', true)` attribute).

**Describe alternatives you've considered**
Add manually a security field in `@@openapi.meta` attribute.
E.g:
```
    @@openapi.meta({
        description: 'A user of the pet store',
        create: {
            description: 'I would like to disable Swagger UI authentication for this operation',
            security: []   <--- here
        },
    })
```

**Additional context**
1. When I add this in the generated specification :
<img width="691" alt="Capture d’écran 2023-03-31 à 15 12 12" src="https://user-images.githubusercontent.com/43639468/229131681-826a760d-0b97-4008-9bca-7fdc9a096403.png">

2. It removes the authentication check :
![Capture d’écran 2023-03-31 à 15 12 28](https://user-images.githubusercontent.com/43639468/229131945-494a47ff-4bdc-44d6-a434-670156370567.png)

Working example : https://github.com/Azzerty23/petstore-openapi-zenstack

**Comment by ymc9** [comment]
Thanks for the proposal @Azzerty23 !

Yes, security settings are missing right now. Following the [specification](https://swagger.io/docs/specification/authentication/), I believe we need to add three things:

1. A way to express security scheme
    Maybe just extend the plugin options:
    ```prisma
    plugin openapi {
        ...
        securityScheme = [
            { myBearer: { type: 'http', scheme: 'bearer' } },
            { myApiKey: { type: 'apiKey', in: 'header', name: 'X-API-KEY' } }
        ]
    }
    ```
    These can just be dumped into the 'components' section in the spec as is.
1. Generate spec using the scheme
    By default, if "securityScheme" is configured, a root-level "security" setting will be added to the spec, including all schemes, 
    ```yaml
    security:
        - myBearer: []
        - myApiKey: []
    ```
    
    On the route level, as you suggested, we can statically analyze if an operation is fully open, and if so, generate an override with `security: []`.
    
1. Explicitly customize route-level security
    As you suggested, we can allow further customization of security for a route using `@@openapi.meta`, either setting it to empty or some specific scheme.

Does this generally look good to you?

**Comment by Azzerty23** [comment]
Yes, that's the perfect solution !

**Comment by Azzerty23** [comment]
I updated the exemple for reference if that helps: https://github.com/Azzerty23/petstore-openapi-zenstack

And while I'm at it, it would have been nice to be able to customize the tags description: `[ModelName] operations` by default. Maybe like this:
```
@@openapi.meta({
        description: 'A user of the pet store',  // to replace User operations
        create: {
            ...
        },
    })
```
![Capture d’écran 2023-04-01 à 11 20 44](https://user-images.githubusercontent.com/43639468/229277658-90022e04-9728-412f-90cd-f84448d349e9.png)

**Comment by ymc9** [comment]
Got it. We'll try to incorporate these in the next release.

**Comment by Azzerty23** [comment]
Again, sorry for that, it would be nice to be able to deprecate an endpoint.

<img width="654" alt="Capture d’écran 2023-04-01 à 11 57 31" src="https://user-images.githubusercontent.com/43639468/229279464-46606fc9-4cc8-4df9-861c-445cc7993431.png">
<img width="644" alt="Capture d’écran 2023-04-01 à 11 58 21" src="https://user-images.githubusercontent.com/43639468/229279473-c2d2fda0-ea35-42cf-a0f8-c5785239510b.png">
<img width="644" alt="Capture d’écran 2023-04-01 à 11 55 36" src="https://user-images.githubusercontent.com/43639468/229279517-34b4ee1b-15de-4609-a967-a7ac01cd24b0.png">

**Comment by ymc9** [comment]
Yes, we can use this issue to keep collecting missing features around openapi.

**Comment by ymc9** [comment]
Fixed by #340 and #342
